### PR TITLE
Fix update for 2020

### PR DESCRIPTION
**Pull request by robtaylor**
Changes for Dubnium and gulp 4. Should make it easier for new contributors.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -180,11 +177,18 @@ gulp.task('serve.https', function() {
...
 // Macro-tasks
-gulp.task('bookmarklets', ['bookmarklet.dev', 'bookmarklet.prod', 'bookmarklet.edge']);
-gulp.task('test', function(next) {
-  seq('browser-test', 'node-test', next);
-});
-gulp.task('work', ['watch', 'serve']);
-gulp.task('https', ['watch', 'serve.https']);
-gulp.task('default', ['lint', 'test', 'build']);
+exports.bookmarklets = gulp.series('bookmarklet.dev', 'bookmarklet.prod', 'bookmarklet.edge');
+test = gulp.series('browser-test', 'node-test');
```
Comment (Yomguithereal) [inline gulpfile.js:189]: Why this global variable?

File: package.json [hunk 2]
```diff
@@ -28,20 +28,22 @@
...
     "gulp-rename": "~1.2.0",
     "gulp-replace": "~0.5.0",
     "gulp-uglify": "^1.1.0",
     "gulp-webserver": "^0.9.0",
     "run-sequence": "^1.0.2"
   },
   "dependencies": {
-    "cheerio": "^0.22.0"
+    "cheerio": "^0.22.0",
+    "natives": "^1.1.6",
```
Comment (Yomguithereal) [inline package.json:46]: What is this?

File: package.json [hunk 3]
```diff
@@ -28,20 +28,22 @@
...
     "gulp-header": "^1.7.1",
     "gulp-jshint": "^1.11.2",
     "gulp-mocha": "^2.0.0",
-    "gulp-mocha-phantomjs": "^0.10.1",
+    "gulp-mocha-phantomjs": "^0.12.1",
     "gulp-rename": "~1.2.0",
     "gulp-replace": "~0.5.0",
     "gulp-uglify": "^1.1.0",
     "gulp-webserver": "^0.9.0",
     "run-sequence": "^1.0.2"
```
Comment (Yomguithereal) [inline package.json:42]: Shouldn't you drop this dependency since you don't require it anymore?

File: package.json [hunk 4]
```diff
@@ -28,20 +28,22 @@
...
   },
   "dependencies": {
-    "cheerio": "^0.22.0"
+    "cheerio": "^0.22.0",
+    "natives": "^1.1.6",
+    "phantomjs": "^2.1.7"
```
Comment (Yomguithereal) [inline package.json:47]: Must this be a dependency? Shouldn't this be in devDeps?

**Review by Yomguithereal — COMMENTED** [review]
Hello @robtaylor. Thanks for the PR. Can you tell me which version of node is dubnium? Also here are some remarks on the code itself.

File: gulpfile.js (hunk 1 above)
Comment (robtaylor) [reply gulpfile.js:189]: do you mean 'test'?

File: package.json (hunk 4 above)
Comment (robtaylor) [reply package.json:47]: good point!

**Comment by robtaylor** [comment]
Dubnium is node 10 LTS

File: gulpfile.js (hunk 1 above)
Comment (Yomguithereal) [reply gulpfile.js:189]: Yes, why is `test` a global variable here?

Comment (robtaylor) [reply gulpfile.js:189]: Its used just below in exports.default

Comment (Yomguithereal) [reply gulpfile.js:189]: Yes, I get it. What I don't get is why you make it a global variable. Haven't you forgotten a `var` here or is this intended?

File: package.json (hunk 2 above)
Comment (robtaylor) [reply package.json]: not needed, it turns out.

File: gulpfile.js (hunk 1 above)
Comment (robtaylor) [reply gulpfile.js:189]: test is used a few lines below for exports.default

File: package.json (hunk 4 above)
Comment (robtaylor) [reply package.json]: good point!

File: package.json (hunk 3 above)
Comment (robtaylor) [reply package.json]: yes!

File: gulpfile.js (hunk 1 above)
Comment (Yomguithereal) [reply gulpfile.js:189]: It should be `var test = gulp.series('browser-test', 'node-test');` nonetheless. The same can be said for `build` line 125 :)

Comment (robtaylor) [reply gulpfile.js]: Sorry, commenting went a bit odd! Got you, should be fixed now.

**Comment by Yomguithereal** [comment]
Thanks for the modifications @robtaylor. Merging now.

**Pull request merged**